src/runtime_functions.js

```diff
@@ -124,6 +124,16 @@ function addFunctionWasm(func, sig) {
   }
 
   // It's not in the table, add it now.
+
+#if ASSERTIONS
+  // Make sure functionsInTableMap is actually up to date, that is, that this
+  // function is not actually in the wasm Table despite not being tracked in
+  // functionsInTableMap.
+  for (var i = 0; i < table.length; i++) {
+    assert(table.get(i) != func, 'function in Table but not functionsInTableMap');
+  }
```

Comment:
Even for a debug build isn't that adding a huge amount of overhead for every function we at the table?

Reply:
Some large projects have tens of thousands of table entries, right?

Reply:
It is quite heavy, yeah. Maybe more than the usual assertions we add. I don't see any noticeable change on CI times, so I *think* this is worth it. It would help debug things like #10899

Another possibility is to add a new option for this, or only do it in `ASSERTIONS=2`.

Reply:
I guess we can land it and revert if we see effects on large projects (I guess it would most effect load time for programs with a lot of function address shared across module boundaries). @awtcode, I'm looking at you :)

Reply:
Thinking some more, this `O(numAddedFunctions*tableSize)` may be pathological sometimes. Let's do it in `ASSERTIONS=2` only? I updated with that.

Reply:
@sbc100 , I think its ok if we only do it for `ASSERTIONS=2`.